v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/cassandra/mappings/CassandraMappings.java

```diff
@@ -40,20 +42,72 @@ public abstract static class Builder {
 
     abstract ImmutableMap.Builder<String, CassandraFieldMapper<?>> fieldMappingBuilder();
 
+    /**
+     * Maintain mappings for a given type, as primitive as well as part of collections.
+     *
+     * @param cassandraType - name of the cassandra type, as discovered by the schema discovery.
+     * @param type - Unified mapping type.
+     * @param rowValueExtractor - {@link CassandraRowValueExtractor} to extract value from {@link
+     *     com.datastax.driver.core.Row Cassandra Row}
+     * @param rowValueMapper - {@link CassandraRowValueMapper} to map value to {@link
+     *     com.google.cloud.teleport.v2.source.reader.io.row.SourceRow}
+     * @param typeClass - Class of the extracted value. Generally return type of the
+     *     rowValueExtractor.
+     * @return Builder
+     */
     public <T> Builder put(
         String cassandraType,
         UnifiedMappingProvider.Type type,
         CassandraRowValueExtractor<T> rowValueExtractor,
-        CassandraRowValueMapper<T> rowValueMapper) {
+        CassandraRowValueMapper<T> rowValueMapper,
+        Class<T> typeClass) {
       this.typeMappingBuilder()
           .put(cassandraType.toUpperCase(), UnifiedMappingProvider.getMapping(type));
       this.fieldMappingBuilder()
           .put(
               cassandraType.toUpperCase(),
               CassandraFieldMapper.create(rowValueExtractor, rowValueMapper));
+      if (!type.equals(UnifiedMappingProvider.Type.UNSUPPORTED)) {
+        putList(cassandraType, type, rowValueExtractor, rowValueMapper, typeClass);
+        putSet(cassandraType, type, rowValueExtractor, rowValueMapper, typeClass);
```

Comment:
Is this right? We do both putSet and putList without and checks?

Reply:
That's right, 
We create `List<Foo>` and a `Set<Foo>` mapping or each scaler type `Foo` as a part of creating a schema and value mapping for the scalar type `Foo`.